addons/mail/static/src/js/models/messages/message.js

```diff
@@ -174,7 +174,10 @@ var Message =  AbstractMessage.extend(Mixins.EventDispatcherMixin, ServicesMixin
         var model = this.getDocumentModel();
         var id = this.getDocumentID();
         var documentThread = this.call('mail_service', 'getDocumentThread', model, id);
-        if (documentThread) {
+        if (documentThread && documentThread._detached) {
```

Comment:
It does not seem to be the correct IMP: with this, any message linked to any document will display `""` as their document name if the document chat window is detached. This means that if I have a chat window with a channel containing messages from documents, e.g. `#RD Tasks`, then it would hide the document link if I have the document thread that is detached.

In fact, I think this does not make sense for the model "message" to provide different document names based on the state of chat windows. The chat window itself should adapt its rendering options to hide the document link.

That means the correct IMP consists of letting thread window pass the option `documentLink` to `false` to the thread widget if the thread is a document thread.

Reply:
changes done.

Reply:
Looks much better! :)